scene/gui/line_edit.cpp

```diff
@@ -45,6 +45,84 @@
 #include "editor/editor_settings.h"
 #endif
 
+void LineEdit::grab_focus_edit(bool p_select_all) {
+	if (!is_inside_tree()) {
+		return;
+	}
+	set_focus_edit(true);
+
+	if (p_select_all) {
+		select_all();
+		set_caret_column(text.length());
+	}
+}
+
+void LineEdit::set_focus_edit(bool p_focus_edit) {
+	if (!is_inside_tree()) {
+		return;
+	}
+
+	if (!has_focus()) {
+		grab_focus();
+	}
+
+	if (focus_edit == p_focus_edit) {
+		return;
+	}
+
+	if (p_focus_edit && !editable) {
+		return;
+	}
+
+	focus_edit = p_focus_edit;
+	emit_signal(SNAME("focus_edit_toggled"), focus_edit);
```

Comment:
This signal is emitted when calling the method, which is something we usually try to avoid. I think it can be handled by connecting to `focus_entered` and checking whether the node is being edited.
Though if it simplifies things, maybe it's justified in this case.